idempotent_interceptor.go

```diff
@@ -18,29 +18,25 @@ type IdempotencyProvider interface {
 	// GetVolumeName should return the name of the volume specified
 	// by the provided volume ID. If the volume does not exist then
 	// an empty string should be returned.
-	GetVolumeName(
-		ctx context.Context,
-		id *csi.VolumeID) (string, error)
+	GetVolumeName(ctx context.Context, id string) (string, error)
```

Comment:
FWIW, I don't think it is necessary anymore to convert an ID to a name, since the ID is now a string, we can use the ID as a map key. Should remove a step from the process...

Reply:
Hi @codenrhoden,

No, this is still necessary due to the way the idempotent interceptor behaves. The `GetVolumeName` function is used by all RPCs that are **not** `CreateVolume` in order to maintain a single concurrency domain based on the volume name. Otherwise  `CreateVolume` would be unable to participate in the concurrency domain as a volume has no ID prior to creation.